tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_consolidation_request.py

```diff
@@ -1020,6 +1020,41 @@ def test_incorrect_source_has_pending_withdrawal(spec, state):
     yield from run_consolidation_processing(spec, state, consolidation, success=False)
 
 
+@with_electra_and_later
+@with_presets([MINIMAL], "need sufficient consolidation churn limit")
+@with_custom_state(
+    balances_fn=scaled_churn_balances_exceed_activation_exit_churn_limit,
+    threshold_fn=default_activation_threshold,
+)
+@spec_test
+@single_phase
+def test_incorrect_source_is_target(spec, state):
+    # move state forward SHARD_COMMITTEE_PERIOD epochs to allow for consolidation
+    state.slot += spec.config.SHARD_COMMITTEE_PERIOD * spec.SLOTS_PER_EPOCH
+
+    # Set up a consolidation with target == source
+    current_epoch = spec.get_current_epoch(state)
+    target_index = source_index = spec.get_active_validator_indices(state, current_epoch)[0]
+    source_address = b"\x22" * 20
+    excess_balance = spec.EFFECTIVE_BALANCE_INCREMENT // 4
+    set_eth1_withdrawal_credential_with_balance(
```

Comment:
I think in this case `set_compounding_withdrawal_credential_with_balance` for target would be enough as the source and target are the same